library/src/main/java/org/apache/apex/malhar/lib/state/managed/Bucket.java

```diff
@@ -389,7 +389,7 @@ public void put(Slice key, long timeBucket, Slice value)
         sizeInBytes.getAndAdd(key.length);
         sizeInBytes.getAndAdd(Long.SIZE);
       }
-      if (timeBucket > bucketedValue.getTimeBucket()) {
+      if (timeBucket >= bucketedValue.getTimeBucket()) {
```

Comment:
For ManagedTimeUnifiedStateImpl, `timeBucket = bucketedValue.getTimeBucket()`. 
For ManagedTimeStateImpl, receiving out of order events is common.
I think we should log a warning.